integration_tests/src/main/python/date_time_test.py

```diff
@@ -671,3 +671,14 @@ def test_timestamp_millis_long_overflow():
 def test_timestamp_micros(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
         lambda spark : unary_op_df(spark, data_gen).selectExpr("timestamp_micros(a)"))
+
+
+@pytest.mark.parametrize('parser_policy', ["LEGACY", "CORRECTED", "EXCEPTION"], ids=idfn)
+def test_date_to_timestamp(parser_policy):
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark : unary_op_df(spark, date_gen),
+        "tab",
+        "SELECT cast(a as timestamp) from tab",
+        conf = {
+            "spark.sql.legacy.timeParserPolicy": "{}".format(parser_policy),
+            "spark.rapids.sql.incompatibleDateFormats.enabled": True})
```

Comment:
Seems we only need `spark.rapids.sql.incompatibleDateFormats.enabled` for `LEGACY` policy?

Reply:
Yes.